crates/sui/src/sui_commands.rs

```diff
@@ -1142,3 +1155,19 @@ fn read_line() -> Result<String, anyhow::Error> {
     io::stdin().read_line(&mut s)?;
     Ok(s.trim_end().to_string())
 }
+
+pub fn parse_host_port(input: Option<String>, port: u16) -> Result<SocketAddr, AddrParseError> {
```

Comment:
- nit: can we call it `default_port`? It's a little confusing to follow at the moment because I keep thinking this is the port being parsed.
- Does the host have to be a socket address (meaning does it need to have an IP address on the LHS, or can someone say `localhost:1234`, for example)?
- I think the default address here should be `0.0.0.0` so that these services can accept connections from interfaces other than the loopback by default.
- Does this function ever get called with `input = None`? I would have thought not, anymore, in which case, let's make `input` non-`Option`-al.
- When do we encounter the `input.is_empty()` case -- is that someone writing `--with-indexer=`? It may be better to treat that as an error rather than interpret it the same as `--with-indexer`, because they may have written `--with-indexer= 1234` or something, and we don't want to silently ignore the port assignment. This may require this function to return an `anyhow::Result` so you can return an expressive error message for this case.

Reply:
> can we call it default_port? It's a little confusing to follow at the moment because I keep thinking this is the port being parsed.

Fair point. The issue was that each service needs a different port, so we need to pass the port for each service as we do not know if the input from the command line has a port or not yet.

> Does the host have to be a socket address (meaning does it need to have an IP address on the LHS, or can someone say localhost:1234, for example)?

Address, yes. We can probably allow `localhost` too!

> I think the default address here should be 0.0.0.0

Sounds good

> When do we encounter the input.is_empty() case -- is that someone writing --with-indexer=? It may be better to treat that as an error rather than interpret it the same as --with-indexer, because they may have written --with-indexer= 1234 or something, and we don't want to silently ignore the port assignment. This may require this function to return an anyhow::Result so you can return an expressive error message for this case.

I think this is now allowed by clap anyway, but it is an extra check for us. For example, using `with-indexer= 1235` (note the space after =) clap will complain.